keps/sig-node/585-runtime-class/README.md

```diff
@@ -0,0 +1,776 @@
+are still side-stepping the issue of precisely defining specific types of runtimes like
+"Sandboxed". However, it is still up for debate whether precisely defining such runtime categories
+is even possible. RuntimeClass allows us to decouple this specification from the implementation, but
+it is still something I hope we can address in a future iteration through the concept of pre-defined
+or "conformant" RuntimeClasses.
+
+**Non-portability.** We are already in a world of non-portability for many features (see [examples
+of runtime variation](#examples-of-runtime-variation). Future improvements to RuntimeClass can help
+address this issue by formally declaring supported features, or matching the runtime that supports a
+given workload
+mitaclly. Another issue is that pods need to refer to a RuntimeClass by name,
+which may not be defined in every cluster. This is something that can be addressed through
+pre-defined runtime classes (see previous risk), and/or by "fitting" pod requirements to compatible
+RuntimeClasses.
+
+## RuntimeClass Scheduling
+
+RuntimeClass scheduling enables native support for heterogeneous clusters where
+every node does not necessarily support every RuntimeClass. This feature allows
+pod authors to select a RuntimeClass without needing to worry about cluster
+topology.
+
+### RuntimeClass Scheduling Motivation
+
+In the initial RuntimeClass implementation, we explicitly assumed that the
+cluster nodes were homogenous with regards to RuntimeClasses. It was still
+possible to run a heterogeneous cluster, but pod authors would need to set
+appropriate [NodeSelector][] rules and [tolerations][taint-and-toleration] to
+ensure the pods landed on supporting nodes.
+
+As [use cases](#runtimeclass-scheduling -user-stories) have appeared and solidified,
+it has become clear that heterogeneous clusters will not be uncommmon, and supporting
+a smoother user experience will be valuable.
+
+[NodeSelector]: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/
+[taint-and-toleration]: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
+
+#### RuntimeClass Scheduling Goals
+
+- Pods using a RuntimeClass that is not supported by all nodes in a cluster are
+  automatically scheduled to nodes that support that RuntimeClass.
+- RuntimeClass scheduling is compatible with other scheduling constraints. For
+  example, a pod with a node selector for GPUs and a Linux runtime should be
+  scheduled to a linux node with GPUs (an intersection).
+
+#### RuntimeClass Scheduling Non-Goals
+
+- Replacing [SchedulingPolicy](#scheduling-policy) with RuntimeClasses.
+
+The following are **currently** out of scope, but _may_ be revisited at a later
+date.
+
+- Automatic topology discovery or node labeling
+- Automatically selecting a RuntimeClass for a pod based on node availability.
+- Defining official or reserved label or taint schemas or for RuntimeClasses.
```

Comment:
@SergeyKanzhelev  - Apologies for the late review but do you think we can add a reserved label for OS so that pod clearly states I am a windows pod vs linux pod?

Reply:
this is in non-goals section. And I kept it this way for this graduation effort.

@tallclair what's the history of making this non-goal? Do you know if there is any other effort for labels reservation?